### PR TITLE
#159951817 Users should be able Signup/Register to Authors Haven

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### What does this PR do?
To be able to use the application and access authenticated routes, Users should be able to Signup and or Register using Sign up forms on the signup route. Therefore, this PR adds a Signup form where users can register onto Authors Haven.

#### What are the relevant pivotal tracker stories?
[#159951817](https://www.pivotaltracker.com/story/show/159951817)

#### Screenshots
screen shot 2018-09-07 at 14 09 33

### CODE REVIEW

**Review by phillipseryazi — APPROVED** [review]

File: src/components/Spinner/Spinner.css [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+@keyframes load8 {
+  0% {
+    -webkit-transform: rotate(0deg);
+    transform: rotate(0deg);
+  }
+  100% {
+    -webkit-transform: rotate(360deg);
+    transform: rotate(360deg);
+  }
+}
```
Comment (geofrocker) [inline src/components/Spinner/Spinner.css]: EOF missing

**Review by geofrocker — COMMENTED** [review]
Resolve merge conflicts

**Review by ssewilliam — CHANGES_REQUESTED** [review]
@AGMETEOR ,
This PR is failing on Heroku,
https://authors-haven-front-pr-14.herokuapp.com/

**Comment by ssewilliam** [comment]
@AGMETEOR 
- the APP crashes when I try to register

screen shot 2018-09-07 at 17 23 18

**Review by ssewilliam — DISMISSED** [review]

**Review by ssewilliam — DISMISSED** [review]

**Comment by AGMETEOR** [comment]
@ssewilliam I have rectified the issue. The URLs were still using localhost and not Heroku.

**Comment by AGMETEOR** [comment]
@ssewilliam I had missed the callback URL in the request data and so redux was dispatching a fail action. The state was therefore not being changed and consequently render the right message. I have fixed it now.

**Review by ssewilliam — APPROVED** [review]

File: src/store/actions/registerActions.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import axios from 'axios';
+import * as actionTypes from './actionTypes';
```
Comment (abulojoshua1) [inline src/store/actions/registerActions.js]: Please remove these wildcards @AGMETEOR.

**Review by abulojoshua1 — CHANGES_REQUESTED** [review]

File: src/store/actions/registerActions.js (hunk 2 above)
Comment (AGMETEOR) [reply src/store/actions/registerActions.js]: @abulojoshua1 let me take them out.

**Comment by geofrocker** [comment]
Use an alert for this message and clear the fields after successful registration
screen shot 2018-09-10 at 2 18 19 pm

**Review by abulojoshua1 — APPROVED** [review]

**Comment by AGMETEOR** [comment]
@geofrocker I have finished the changes.

**Comment by geofrocker** [comment]
Cool

**Pull request merged**